dawn/test/unit-test/dawn/Optimizer/TestPassFieldVersioning.cpp

```diff
@@ -199,4 +206,71 @@ TEST_F(TestPassFieldVersioning, VersioningTest7) {
   int idA = instantiation->getMetaData().getAccessIDFromName("field_a");
   ASSERT_TRUE(instantiation->getMetaData().isMultiVersionedField(idA));
 }
+
+TEST_F(TestPassFieldVersioning, VersionSparseField) {
+  // when a sparse field is fixed due to double buffering it needs to be filled. for sparse fields,
+  // this fill requires a loop statement to be generated
+
+  using namespace dawn::iir;
+  using LocType = dawn::ast::LocationType;
+
+  UnstructuredIIRBuilder b;
+  auto dense = b.field("dense", LocType::Edges);
+  auto sparse =
+      b.field("sparse", {LocType::Edges, LocType::Cells, LocType::Vertices, LocType::Edges});
+
+  auto stencil = b.build(
+      "OffsetReadsInCorrectContext",
+      b.stencil(b.multistage(
+          LoopOrderKind::Parallel,
+          b.stage(b.doMethod(
+              dawn::sir::Interval::Start, dawn::sir::Interval::End,
+              b.stmt(b.assignExpr(b.at(dense),
+                                  b.reduceOverNeighborExpr(Op::plus, b.at(sparse), b.lit(0.),
+                                                           {LocType::Edges, LocType::Cells,
+                                                            LocType::Vertices, LocType::Edges}))),
+              b.loopStmtChain(
+                  b.stmt(b.assignExpr(b.at(sparse), b.at(dense, HOffsetType::noOffset, 0),
+                                      Op::assign)),
+                  {LocType::Edges, LocType::Cells, LocType::Vertices, LocType::Edges}))))));
```

Comment:
This test is currently testing a transformation that shouldn't happen. When field versioning of sparse fields will be fixed this should maybe become a negative test. I suggest commenting it out and have the following positive test instead:
```
dense_field_edges = reduce(
  Edges > Verts > Edges,
  reduce(
    Edges > Cells,
    sparse_field_e_c
  )
)
with sparse[Edges > Cells]:
  sparse_field_e_c = dense_field_edges[Edges]
```
(much can be reused from the code below i think)